crates/wac-graph/src/plug.rs

```diff
@@ -0,0 +1,64 @@
+use crate::{
+    types::{Package, SubtypeChecker},
+    CompositionGraph, EncodeOptions, NodeId, PackageId,
+};
+
+/// Take the exports of the plug component and plug them into the socket component.
+pub fn plug(
+    graph: &mut CompositionGraph,
+    plugs: Vec<Package>,
+    socket: PackageId,
+    socket_instantiation: NodeId,
```

Comment:
Would it be possible to do `graph.instantiate(socket);` inside this function instead of needing to do it outside and pass in the `socket_instantiation`. That would reduce the number of arguments and hopefully make this easier to understand.